### PR TITLE
[4.3] Workflow Transitions

### PR DESCRIPTION
**Pull request by brianteeman**
Recently there have been several people posting that they don't understand Workflows. I was reviewing my notes from the Workflows presentation I gave this year at JoomlaDay fr and I realised that we can easily remove one area of confusion.

### Summary of Changes
Add a title to the options so that it is clear what they are for. It was confusing before because the displayed value is a stage but the options are a transition. 

Bizaarely we must have realised this before because the Action button already adds some clarification by saying "Run Transitions"
image

**NOTE: If you are using the workflow from the sample data the problems this PR fixes are more obvious**

### Testing Instructions
1. Enable workflows
2. in the article list view click on the stage
3. in the article edit view click on the workflow

### Actual result BEFORE applying this Pull Request
In the list view it is not clear that there are two different things in the list - current stage name and transitions
image

In the article view it says Workflow but it should say Workflow Stage and it is not clear that there are two different things in the list - current stage name and transitions
image

### Expected result AFTER applying this Pull Request

Article List View
image

Article Edit View
image

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
screenshots will need updating

- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on cf92a8d226b1f65b11f77cd97c0f4f4f2b4565f8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39325">issues.joomla.org/tracker/joomla-cms/39325</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on cf92a8d226b1f65b11f77cd97c0f4f4f2b4565f8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39325">issues.joomla.org/tracker/joomla-cms/39325</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39325">issues.joomla.org/tracker/joomla-cms/39325</a>.</sub>

**Comment by obuisard** [comment]
I did not catch this one in time, but the fields could have been modified to extend GroupediistField rather than ListField, where grouping of options is handled more elegantly and would prevent the hard coding of <OPTGROUP>

**Comment by brianteeman** [comment]
Do you want me to look at updating this pr with that?

**Comment by brianteeman** [comment]
will do boss

**Comment by brianteeman** [comment]
Having looked at the code I am going to pass on making the change myself and stick with my pr. Feel free of course to reject it and to submit your own.

**Comment by obuisard** [comment]
> Having looked at the code I am going to pass on making the change myself and stick with my pr. Feel free of course to reject it and to submit your own.

@brianteeman I do not want to close this PR because of the work that you put into it and the user benefits. 
However, I still think the TransitionField.php should be updated. 
There are 5 lines of code to modify and 2 to remove.
I can modify your file and commit the changes so we can save ourselves some time. Ok with you?

**Comment by brianteeman** [comment]
I would prefer it if you did it as a pr to my branch

**Comment by obuisard** [comment]
> I would prefer it if you did it as a pr to my branch

Ok, I can do that.

**Comment by brianteeman** [comment]
This will need you tests

**Comment by obuisard** [comment]
I have tested this item :white_check_mark: successfully on 9cbbc25db62e042eb1b3478757ef5b546d159411<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39325">issues.joomla.org/tracker/joomla-cms/39325</a>.</sub>

**Comment by obuisard** [comment]
Jacob @jwaisner, would you mind testing the PR once more as the Transition field has been updated to extend Groupedlist? Thank you very much!

**Comment by obuisard** [comment]
Vio @viocassel, we need a new test for this updated PR, do you mind re-testing? Thank you!

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 9cbbc25db62e042eb1b3478757ef5b546d159411<br /><br />👍<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39325">issues.joomla.org/tracker/joomla-cms/39325</a>.</sub>

File: administrator/language/en-GB/com_content.ini [hunk 1]
```diff
@@ -189,6 +190,7 @@ COM_CONTENT_URL_FIELD_BROWSERNAV_LABEL="URL Target Window"
 COM_CONTENT_URL_FIELD_C_BROWSERNAV_LABEL="URL C Target Window"
 COM_CONTENT_WARNING_PROVIDE_VALID_NAME="Please provide a valid, non-blank title."
 COM_CONTENT_WORKFLOW="Workflow"
+COM_CONTENT_WORKFLOW_STAGE="Workflow Stage"
 COM_CONTENT_WORKFLOW_NOT_FOUND="No default workflow available, please define one or contact an administrator."
```
Comment (Quy) [inline administrator/language/en-GB/com_content.ini]:
```suggestion
COM_CONTENT_WORKFLOW_NOT_FOUND="No default workflow available, please define one or contact an administrator."
COM_CONTENT_WORKFLOW_STAGE="Workflow Stage"
```

Comment (obuisard) [reply administrator/language/en-GB/com_content.ini]: Good catch!

**Comment by obuisard** [comment]
Thank you Brian @brianteeman !

**Comment by brianteeman** [comment]
thx

**Comment by Quy** [comment]
@brianteeman Can you please look into this for the frontend? Thanks.

39325-workflow

**Pull request merged**